Source/WebInspectorUI/UserInterface/Models/Resource.js

```diff
@@ -435,6 +435,14 @@ WI.Resource = class Resource extends WI.SourceCode
         return this._mimeType;
     }
 
+    get hasMetadata()
+    {
+        // Information collected about resources loaded before Web Inspector was opened is missing most metadata,
+        // like resource timing data, HTTP method, request headers, etc.
+        // Missing a requestSentDate is an indicator that other important metadata is also missing.
```

Comment:
these sentences are kinda awkward to read.  i'd reword them something along the lines of
```
// Some metadata is only collected when Web Inspector is open (e.g. resource timing data, HTTP method, request headers, etc.).
// Use `requestSentDate` as a general signal since it is always included when metadata is collected.
```

also, i'm not sure this was provided before iOS 11 (see `WI.NetworkObserver.prototype.requestWillBeSent` so we might need to find a different general signal 😕  maybe `requestIdentifier`?

Reply:
Updated. `requestIdentifier` works as a signal